tests/cases/compiler/selfReferentialFunctionType.ts

```diff
@@ -0,0 +1,8 @@
+// @strict: true
+// @declaration: true
+
+function f<T>(args: typeof f<T>): T { return args; }
+
+function g<T = typeof g>(args: T): T { return args; }
+
+function h<T>(): typeof h<T> { return h; }
```

Comment:
I know I wrote these tests, but on second thought, it's probably wiser to write
```suggestion
declare function f<T>(args: typeof f<T>): T;

declare function g<T = typeof g>(args: T): T;

declare function h<T>(): typeof h<T>;
```

And then they'll be error-free.